app/ui-react/packages/ui/src/Integration/Editor/template/TemplateStepTypeSelector.tsx

```diff
@@ -1,11 +1,12 @@
 import { Text, TextContent } from '@patternfly/react-core';
 import * as React from 'react';
+import {
+  LintFreemarker,
+  LintMustache,
+  LintVelocity,
+} from '../../../Shared/TextEditor';
 
-export enum TemplateType {
-  Freemarker = 'freemarker',
-  Mustache = 'mustache',
-  Velocity = 'velocity',
-}
+export type TemplateType = LintFreemarker | LintMustache | LintVelocity;
```

Comment:
I dunno, the problem then is we've special strings, I kinda thought the enums were a little more helpful.

Reply:
The problem with the enum is that then the linter class is coupled with the template step mode selector. If we need a second specialized editor with different modes we'd need a second enum, but then which one should the linter class use?

This way it's a bit less type-safe because we are just enriching Codemirror modes with new modes, and delegate the control of the proper mode selection to whatever systems will be used to do that (radios for the templater)

Reply:
yeah, good point, also I don't feel strongly enough to worry about it too much either :smile: